packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/VirtualizedTraceView.tsx

```diff
@@ -223,6 +202,28 @@ export class VirtualizedTraceViewImpl extends React.Component<VirtualizedTraceVi
     }
   }
 
+  getRowStates(): RowState[] {
+    const { childrenHiddenIDs, detailStates, trace } = this.props;
+    return memoizedGenerateRowStates(trace, childrenHiddenIDs, detailStates);
+  }
+
+  getClippingCssClasses(): string {
+    const { currentViewRangeTime } = this.props;
+    return memoizedGetCssClasses(currentViewRangeTime);
```

Comment:
Is `currentViewRangeTime` stable? Or the object is created in every render?

Reply:
Same as above.